cmd/bnsd/app/app_performance_test.go

```diff
@@ -159,34 +167,33 @@ func BenchmarkBNSDSendToken(b *testing.B) {
 						},
 					},
 				}
-				// hmmm.... can we collapse this to the message and one line to get nonce and sign?
-				nonce, err := aliceNonce.Next()
-				if err != nil {
-					b.Fatalf("getting nonce failed with %+v", err)
-				}
-				sig, err := sigs.SignTx(aliceKey, tx, "mychain", nonce)
+				sig, err := sigs.SignTx(aliceKey, tx, "mychain", aliceNonce)
 				if err != nil {
 					b.Fatalf("cannot sign transaction %+v", err)
 				}
 				tx.Signatures = append(tx.Signatures, sig)
 				txs[k] = tx
 
-				// must reset nonce per block for CheckOnly
 				if !tc.strategy.Has(weavetest.ExecDeliver) && (k+1)%tc.txPerBlock == 0 {
-					aliceNonce = NewNonce(runner, alice)
+					// When the transaction is split into blocks and the previous block
```

Comment:
Yeah, this is a bit confusing but well documented here.
Let's think of a cleaner solution for the future